x-pack/plugins/security_solution/public/management/cypress/tasks/fleet.ts

```diff
@@ -83,3 +87,131 @@ export const createAgentPolicyTask = (
+      agent_features: [{ name: 'tamper_protection', enabled: true }], // TODO: this can be removed once FF code is removed
+      is_protected: true,
+      description: 'test',
+      namespace: 'default',
+      monitoring_enabled: ['logs', 'metrics'],
+      inactivity_timeout: 1209600,
+    },
+    headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+  });
+};
+
+export const getUninstallToken = (agentPolicyId: string) => {
+  return request<GetUninstallTokensMetadataResponse>({
+    method: 'GET',
+    url: `${uninstallTokensRouteService.getListPath()}?policyId=${agentPolicyId}`,
+    headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+  }).then((uninstallTokenResponse) => {
+    return request<{ item: UninstallToken }>({
+      method: 'GET',
+      url: uninstallTokensRouteService.getInfoPath(uninstallTokenResponse.body.items[0].id),
+      headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+    });
+  });
+};
+
+export const unenrollAgent = (agentId: string): Cypress.Chainable<boolean> => {
+  return request({
+    method: 'POST',
+    url: agentRouteService.getUnenrollPath(agentId),
+    headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+  }).then(() => {
+    return isAgentUnenrolled(agentId);
+  });
+};
+
+export const changeAgentPolicy = (agentId: string, policyId: string) => {
+  return request({
+    method: 'POST',
+    url: agentRouteService.getReassignPath(agentId),
+    body: {
+      policy_id: policyId,
+    },
+    headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+  }).then(() => {
+    return hasAgentPolicyChanged(agentId);
+  });
+};
+
+// only used in "real" endpoint tests not in mocked ones
+export const uninstallAgentFromHost = (
+  hostname: string,
+  uninstallToken?: string
+): Cypress.Chainable<string> => {
+  return cy.task('uninstallAgentFromHost', {
+    hostname,
+    uninstallToken,
+  });
+};
+
+// only used in "real" endpoint tests not in mocked ones
+export const isAgentAndEndpointUninstalledFromHost = (
+  hostname: string
+): Cypress.Chainable<boolean> => {
+  return cy.task('isAgentAndEndpointUninstalledFromHost', {
+    hostname,
+  });
+};
+
+const isAgentUnenrolled = (agentId: string): Cypress.Chainable<boolean> => {
+  let isUnenrolled = false;
+  return cy
+    .waitUntil(
+      () => {
+        return request<GetOneAgentResponse>({
+          method: 'GET',
+          url: agentRouteService.getInfoPath(agentId),
+          headers: {
+            'elastic-api-version': API_VERSIONS.public.v1,
+          },
+        }).then((response) => {
+          if (response.body.item.status === 'unenrolled' && !response.body.item.active) {
+            isUnenrolled = true;
+            return true;
+          }
+
+          return false;
+        });
+      },
+      { timeout: 120000 }
+    )
+    .then(() => {
+      return isUnenrolled;
+    });
+};
+
+const hasAgentPolicyChanged = (agentId: string): Cypress.Chainable<boolean> => {
```

Comment:
Name of this function is a bit confusing because its not checking that a policy changed. Its checking if the Agent is in a status other than `updating`.  I don't remember when an agent goes into `updating` - is it only during initial enrollment? or every time a new policy is received by the agent?

If only during initial enrolment, then maybe rename it `isAgentEnrolled()`. 
If it goes into this state on every policy update, then it feels like you need more filtering criteria below and a new input argument - like the Policy Version number that you expected to have been applied. 

let me know if I'm missing something here.

Reply:
I'm trying to ensure agent has been updated after reassigning new policy. You are right that more filtering would be worth here.